modules/cloudflare_delegation/main.tf

```
@@ -0,0 +1,13 @@
+resource "cloudflare_zone" "this" {
+  account_id = var.account_id
+  zone       = var.domain_name
+}
+
+resource "cloudflare_record" "ns" {
+  count   = length(var.aws_route53_name_servers)
```

```suggestion
  count   = length(var.name_servers)
```
because the NS can be anywhere, not just in route53